source/fundamentals/crud/read-operations/sort.txt

```diff
@@ -4,135 +4,396 @@ Sort Results
 
 .. default-domain:: mongodb
 
-Use ``sort()`` to change the order in which read operations return
-documents. The ``sort()`` instance method tells MongoDB to order returned documents by the
-values of one or more fields in a certain direction. To sort returned
-documents by a field in ascending (lowest first) order, use the static method ``Sorts.ascending()``, 
-specifying the field on which to sort. To sort in descending (greatest first) order instead, 
-use the static method ``Sorts.descending()``, specifying the field on which to sort. For example,
-you can sort on ``id`` or ``name`` fields. If you do not specify a sort, MongoDB does not 
-guarantee the order of query results.
-
-Follow the instructions in the examples below to insert data into
-a collection and perform a sort on the results of a query.
-Consider a collection containing documents that describe books. To
-insert this data into a collection, run the following operation:
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, we show you how to **sort** your results.
+
+Sorting orders the documents returned from your query by your specified 
+**sort criteria**. Sort criteria are the rules you give MongoDB that describe
+how you would like your data ordered. Some examples of sort criteria are:
+
+* Smallest number to largest number
+* Earliest time of day to latest time of day
+* Alphabetical order by first name 
+
+You should read this guide if you would like to:
+
+* Perform ascending sorts and descending sorts.
+* Combine sort criteria.
+* Sort on the text score of a
+  :manual:`text search </core/text-search-operators/>`.
+
+.. _sorts-crud-sort-example:
+
+The examples in this guide use a sample collection, ``sort_example``, that
+contains the following documents:
+  
+.. code-block:: json
+  
+   {"_id": 1, "letter": "c", "food": "coffee with milk"}
+   {"_id": 3, "letter": "a", "food": "maple syrup"}
+   {"_id": 4, "letter": "b", "food": "coffee with sugar"}
+   {"_id": 5, "letter": "a", "food": "milk and cookies"}
+   {"_id": 2, "letter": "a", "food": "donuts and coffee"}
+   {"_id": 6, "letter": "c", "food": "maple donut"}
+
+Methods For Sorting
+-------------------
+
+You can sort your results using the  ``sort()`` method of a ``FindIterable``
+instance or by using the  ``Aggregates.sort()`` static factory method in an
+aggregation pipeline. Both of these methods take objects that implement the
+``Bson`` interface and specify sort criteria as arguments. For information
```

Comment:
```suggestion
:ref:`Bson <bson>` interface and specify sort criteria as arguments.
```
The sentence after can be removed.

Reply:
Thank you for the note. I disagree with this suggestion as the link would not provide context to what it is linking to, in addition the suggested link is not linking to the documentation of the BSON interface but to the Bson fundamentals page.

Reply:
Maybe you could link to the javadoc for Bson here instead?

Reply:
Thanks for the suggestion. In a discussion with Chris.C, I was told that as a general rule we are trying to avoid inline Java Doc links and instead group them at the end of the section. This discussions where this took place are linked to in https://github.com/mongodb/docs-java/pull/66#discussion_r601822352. I should have probably listed that as well in my earlier comment in this thread, apologies.